File: src/main/java/com/sy/coladay/security/AuthenticationFacade.java
package com.sy.coladay.security;

import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;

/**
 * Facade that provide easy access to the current <code>UserPrincipal</code>.
 *
 * @author selim
 * @see UserPrincipal
 * @see SecurityContextHolder
 */
@Component
public class AuthenticationFacade {

  /**
   * Return an instance of the current principal authenticated.
   *
   * @return the current <code>UserPrincipal</code> being authenticated.
   */
  public UserPrincipal getCurrentUserPrincipal() {
    return (UserPrincipal) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
  }

}

File: src/main/kotlin/com/sy/coladay/security/AuthenticationFacade.kt
package com.sy.coladay.security

import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

/**
 * Facade that provide easy access to the current `UserPrincipal`.
 *
 * @author selim
 * @see UserPrincipal
 *
 * @see SecurityContextHolder
 */
@Component
class AuthenticationFacade {
    /**
     * Return an instance of the current principal authenticated.
     *
     * @return the current `UserPrincipal` being authenticated.
     */
    fun getCurrentUserPrincipal():UserPrincipal {
        return SecurityContextHolder.getContext().authentication.principal as UserPrincipal
    }
}